aten/src/ATen/FunctionalTensorWrapper.cpp

```diff
@@ -183,9 +185,20 @@ void FunctionalTensorWrapper::replace_(const Tensor& other) {
   value_ = other;
   // out= ops are allowed to resize the output tensors, mutating both the data and metadata of the tensor.
   // We need to propagate that metadata mutation to the wrapper (new size).
-  if (sizes() != value_.sizes() || strides() != value_.strides()) {
-    set_sizes_and_strides(value_.sizes(), value_.strides());
+  //
+  if (!has_symbolic_sizes_strides()) {
+    if (sizes() != value_.sizes() || strides() != value_.strides()) {
+      set_sizes_and_strides(value_.sizes(), value_.strides());
+      refresh_numel();
+      refresh_contiguous();
+    }
+  } else {
+    // For sym ints, this will check pointer equality, which I'm pretty sure is what we want.
```

Comment:
yes. Importantly, the pointer equality test will never falsely say that the sizes are the same when they are not